### PR TITLE
Added use of subdirs for cache items.

### PR DESCRIPTION
**Pull request by ppbrown**
It will split things up in groups of 10,000, so that we dont have 1,000,000 items in a single directory.

Needs testing though.

### CODE REVIEW

**Comment by Arcitec** [comment]
Needs testing, but is a great idea. Segmenting huge amounts of files like this is very common and makes sense to add it here too.

**Comment by Nerogar** [comment]
>  It really needs to be addressed: This code needs to be backwards-compatible so that it can still use "v1" cache directories for everyone who has existing data

Maybe the best strategy would be to migrate existing caches. Adding a full versioning scheme doesn't make much sense. Caches aren't really supposed to be compatible for multiple versions. And a migration function could simply check if 0.pt exists in the base directory.

**Comment by ppbrown** [comment]
I should add that, longer term, I'd also like to see a format that is more friendly for adding entries.
I'm not sure that the current format, of having the "index" be a pytorch file, is the best choice for that?

When I summarized things for chatgpt, and specifically mentioned scaling to 1million+ files, it suggested either sqlite format, or LMDB format instead for the index.

**Comment by Nerogar** [comment]
If you mean the aggregate.pt file, that's not an index. It contains part of the cache that needs to be loaded at all times because it's frequently and randomly accessed.

**Comment by ppbrown** [comment]
err...
you say it isnt an index.
maybe it has additional functionality, but... it ALSO functions as an index.
It maps (original location of source image) to (entrynum.pt file)

thats an index.

**Comment by Nerogar** [comment]
no. It doesn't map anything. As I said, it just saves some part of the cache. The file name is one of those parts. But it also contains things like the original resolution, cropped resolution, and other things depending on your configuration.
The actual mapping is always done using the input of previous modules. For OneTrainer, that's (currently) a sorted list of file names from the original concept directory. This is also the reason why the original files are always needed.

**Comment by ppbrown** [comment]
Maybe we are getting unneccessarily caught up with definitions like "map" and "index", when that isnt the most important thing here.

So let me rephrase my earlier question:
Why are you using a pytorch object for this, when it seems like an actual database object for all the stuff would seem to fit the usage much better?

It sort of seems just like "well i'm using pytorch for this other stuff so I'm going to just go ahead and do the same thing over here", even though it isnt a good fit.

I'm asking if there is a specific reason for it, because otherwise, at some point in the future, I may submit a PR that changes the internal format to one of the other two more efficient formats that I mentioned earlier
(sqlite or LMDB)

**Comment by ppbrown** [comment]
ah.. so now it almost makes sense why its called "aggregate.pt".
Its just a "bunch o stuff".
Thanks for the info.
